### PR TITLE
[`ConvertSlow`] make sure the order is preserved for addedtokens

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
@zucchini-nlp caught a nice bug: the order of the user defined tokens and control tokens was not preserved

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Ok!

File: src/transformers/convert_slow_tokenizer.py [hunk 1]
```diff
@@ -622,16 +622,19 @@ def decoder(self, replacement, add_prefix_space):
...
 
-        tokenizer.add_tokens(user_defined_symbols + control_symbols)
+        all_added_tokens = {
+            id: AddedToken(token, normalized=False, special=special)
+            for id, token, special in [
+                (id, p.piece, p.type == 3) for id, p in enumerate(self.proto.pieces) if p.type in [3, 4]
+            ]
+        }
+
+        tokenizer.add_tokens([k for _, k in sorted(all_added_tokens.items(), key=lambda x: x[0])])
```
Comment (itazap) [inline src/transformers/convert_slow_tokenizer.py:637]: Thanks for the fast fix! Is the sorting here necessary or is it already sorted from enumerate?

Comment (ArthurZucker) [reply src/transformers/convert_slow_tokenizer.py:637]: not sure that the dict is an ordered dict by default for all python version! just for safety!

**Comment by ArthurZucker** [comment]
image
tests are failing because this is the sentencepiece.model that was uploaded

**Comment by ArthurZucker** [comment]
But it's also related to `tokenizers` previously mentioned bug, where you cannot add special tokens with `add_tokens` (main of `tokenizers` does not fail on this)

**Pull request merged**